css/how_to_name_the_files.md

```diff
@@ -0,0 +1,115 @@
+# How to name the files?
+
+To understand which is the best name for the css/scss files
+
+## 1. The underscore character
+
+The first difference that we have to highlight is about the `_` underscore character. It is used at the beginning of the name of those files that don't need to be compiled.
+
+Most of the scss files must have the `_` underscore character, the only exception should be that particular stylesheet that needs to be imported into a different module or part of the application, without the rest of the main css.
+
+---
+
+**SCSS Input**
+
+-   src
+    -   text-editor
+        -   text-editor.scss
+    -   blog
+        -   \_blog.scss
+        -   main.scss
+            ```scss
+            @import blog/_blog.scss @import text-editor/text-editor.scss;
+            ```
+
+**CSS Output**
+
+-   build
+    -   text-editor.scss
+        ```scss
+        // this file contains the text-editor.scss style
+        ```
+    -   main.scss
+        ```scss
+        // this file contains the _blog.scss + text-editor.scss style
+        ```
+
+In this example the code is generating 2 different css and it would be perfect if we had 2 different modules
+
+**HTML Use**
+
+-   module-main.html
+    ```html
+    <link rel="stylesheet" href="main.css" />
+    ```
+-   module-text-editor.html
+    ```html
+    <link rel="stylesheet" href="text-editor.css" />
+    ```
+
+---
+
+But, If you don't need the second generated css, please use the `_` underscore character to avoid the creation of an extra compiled files
+
+**SCSS Input**
+
+-   src
+    -   text-editor
+        -   `_`text-editor.scss
+    -   blog
+        -   \_blog.scss
+        -   main.scss
+            ```scss
+            @import blog/_blog.scss @import text-editor/text-editor.scss;
+            ```
+
+**CSS Output**
+
+-   build
+    -   main.scss
+        ```scss
+        // this file contains the _blog.scss + text-editor.scss styles
+        ```
+
+**HTML Use**
+
+-   module-main.html
+    ```html
+    <link rel="stylesheet" href="main.css" />
+    ```
+
+## 2. User friendly names
+
+Sometimes we need a composed name for our files, example: \_text-editor.scss, \_button-style.scss, \_card-image.scss
+
+The convention defined by Clay and Bootstrap is to use a single `-` hyphen character to create a separation between the words, lowercase Unicode characters to write the names, and the `_` underscore character mentioned in the previous section
+
+To simplify the syncronization process between the frameworks we decided to extend the same pattern to all our projects
+
+So, don't use:
+
+-   \_text_editor.scss
```

Comment:
We do exactly the opposite in `liferay-portal`, because mixing `_` and `-` is not allowed. We need to either recommend this other approach (and just not follow it in Clay) or make a clear note about this.

Reply:
Is there a particular reason why mixing _ and - is not allowed?

Reply:
Brian doesn't like mixing, that's all. Module names have `-` and file names have `_`.

That doesn't mean we have to change this in places like Clay, but we need to note that in DXP, the convention for filenames is different...